libs/wallet/provider/uphold/uphold.go

```diff
@@ -169,7 +181,7 @@ func newRequest(method, path string, body io.Reader) (*http.Request, error) {
 	return req, err
 }
 
-func submit(logger *zerolog.Logger, req *http.Request) ([]byte, *http.Response, error) {
+func (w *Wallet) submit(logger *zerolog.Logger, req *http.Request) ([]byte, *http.Response, error) {
```

Comment:
With these changes, `submit` becomes part of `Wallet` only because of the conditional choice of a client.

I don't think it's enough to justify the promotion of the function to `Wallet`.

Reply:
I refactored the code to pass `*http.Client` to the `submit()` function and then `IsUserKYC()` uses a version of `SubmitTransaction()` that takes an extra `*http.Client`